app/scripts/modules/core/domain/serverGroup.ts

```diff
@@ -12,28 +12,30 @@ interface IAsg {
 }
 
 export interface ServerGroup {
-  account: string;
+  account?: string;
   app?: string;
   asg?: IAsg;
   buildInfo?: any;
   category?: string;
-  cloudProvider: string;
-  cluster: string;
+  cloudProvider?: string;
+  cluster?: string;
   entityTags?: IEntityTags;
   detail?: string;
   executions?: Execution[];
-  instanceCounts: InstanceCounts;
+  instanceCounts?: InstanceCounts;
   instanceType?: string;
-  instances: Instance[];
+  instances?: Instance[];
+  isDisabled?: boolean;
   launchConfig?: any;
   loadBalancers?: string[];
-  name: string;
+  name?: string;
   provider?: string;
-  region: string;
+  region?: string;
   runningTasks?: ITask[];
   searchField?: string;
+  securityGroups?: string[];
   stringVal?: string;
   tags?: any;
-  type: string;
+  type?: string;
   vpcName?: string;
 }
```

Comment:
The problem stems from where we construct servergroups on the fly.  if we don't mark the fields as optional, we have to explicitly set them.  E.g., in `addServerGroupSecurityGroups` we push a new `ServerGroup` onto the security group's usages array for server groups but we only set three properties:  `name`, `isDisabled`, and `region`.  This is one of the bigger downsides of using interface types in TS rather than just model classes.  Thoughts?